application/tests/unit_handler.js

```diff
@@ -525,12 +565,35 @@ describe('User service', () => {
 
     //usergroups
 
+    it('Try to create wrong usergroup', () => {
+      let req = {
+        payload: wrong_usergroup,
+        auth: { //headers which will be set with JWT
+          credentials: {
+            userid: userid
+          }
+        }
+      };
+      return handler.createOrUpdateUsergroup(req, (result) => {
+        // console.log(result);
+
+        expect(result.isBoom).to.equal(true);
+        expect(result.output.statusCode).to.equal(500);
+
+        return;
+      })
+        .catch((Error) => {
```

Comment:
The createOrUpdateUsergroup does NOT generally return a promise, there is actually  a case where it doesn't so it might error then. Also unnecessary if it did always return a promise, as it only logs the error. The mocha tests would catch it and log it more properly, no reason to console.log in tests especially.

Reply:
I like to have the whole error object in the logs (which is not the case if mocha catches it by itself) thus I somewhen started to catch and throw the error while expecting Promise behaviour. I want to not change it because in simple cases the mocha error log is just used which is enough, and in the more complicated cases the additional logging helps out.